docs/integration-playbook.md

```diff
@@ -0,0 +1,188 @@
+to catch mistakes sooner and easily fix any issues it surfaces. Correcting an
+issue in your code editor is much easier than waiting days or weeks for a
+security or QA team to surface the issue. However, CI systems are often more
+reliable and repeatable than results from developer's environments. This
+section will move from earliest to latest in the development pipeline.
+
+**One-off Analyses**
+
+Of course, Semgrep can always be [run locally](https://github.com/returntocorp/semgrep#getting-started)
+to investigate findings, search for new patterns, or simply to tinker with the
+tool. Running one-off analyses provides maximal flexbility while losing
+reliability, repeatablility, and consistency. We recommend using this approach
+during initial investigations and then moving to one of the below approaches
+once the tool has been tuned and [configured](https://github.com/returntocorp/semgrep/blob/develop/docs/configuration-files.md)
+to your environemnt.
+
+**In-editor**
+
+In-editor integration allows you to see Semgrep findings inside your editor.
+This is the fastest way to get Semgrep results and offers the convenience of
+seeing findings alongside the code you're editing.
+
+In-editor integration is available for the following code editors:
+
+* [Visual Studio Code](https://marketplace.visualstudio.com/items?itemName=semgrep.semgrep)
+
+**Git Hooks**
+
+Git hooks integration allows you to check for findings before code is pushed
+to a remote repository. This allows you to quickly see issues without waiting
+for a CI process to run or a code reviewer to notice them.
+
+r2c recommends using the [`pre-commit`](https://pre-commit.com/) framework for
+Git hook integration. However, the following is a complete listing of Git hook
+integrations:
+
+* [`pre-commit`](https://github.com/returntocorp/semgrep/blob/develop/docs/integrations.md#pre-commit-hook)
+* [Git Hooks](https://www.atlassian.com/git/tutorials/git-hooks)
+
+**Continuous Integration**
+
+Continuous Integration (CI) allows you to check each commit for findings and
+halt the run if necessary. CI is typically invoked on a remote server when
+changes are pushed to a [VCS](https://en.wikipedia.org/wiki/Version_control).
+Including Semgrep in CI gives the most assurance that no findings will fall
+through the cracks. CI environments are often more reliable than expecting
+uniformity and consistency across many development environments.
+
+The following is a complete listing of CI integration avenues alongside our
+existing CI provider documentation:
+
+* [Semgrep CI Providers](https://github.com/returntocorp/semgrep/blob/develop/docs/integrations.md#continuous-integration)
+* [`semgrep-action` GitHub Action](https://github.com/marketplace/actions/semgrep-action)
+* [Coinbase Salus](https://github.com/coinbase/salus)
+
+### Security
+
+Security teams will naturally have an adversarial relationship with development.
```

Comment:
This reads pessimistically to me. While I agree that pointing out the friction between devs and appsec can evoke the kind of response we want, I think as written it has the danger of engendering that friction further. How could we rephrase this in such a way that Semgrep presents itself as a solution to the problem of developer/security friction?

Reply:
+1